### DISCUSSION TITLE
Adding delay to input live stream

### DISCUSSION THREAD

**Discussion by gavin2812** [savonet/liquidsoap#3271] [User Support]
Hello,

I was hoping to add a delay to the incoming stream as the metadata seems to be delayed in appearing, so was hoping to do this in order to add a delay to the input stream to be able to sync the on-screen items, currently the metadata is running around 10 seconds behind the audio stream (I get the metadata from an external file):

```
## Set Audio Stream Input
audio_url = input.ffmpeg("http://inputlivestream.com/stream")
audio_delayed = buffer(buffer=10., max=10., audio_url)
final_audio = ffmpeg.decode.audio(audio_delayed)
final_audio = mksafe(final_audio)

```
But this seems to only delay the whole script from running (the delay of 10 seconds is at the beginning when the script starts, not just delaying the audio stream).

Have I misunderstood how this works, maybe wrong implementation or does it now work this way? If I have misunderstood, is there a different way I could accomplish this?

Thanks.

**Comment by toots** [comment]
This should work. How are you getting the metadata? Is it part of the input stream?
